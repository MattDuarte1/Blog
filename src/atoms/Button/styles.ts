import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme, disabled }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 20px 23px 19px;
    gap: 10px;
    border: none;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};

    min-width: 145px;
    height: 67px;
    font-weight: 700;
    color: #ffff;

    background: ${theme.colors.blue};
    border-radius: 8px;
    transition: filter 0.4s ease;

    filter: ${disabled && 'brightness(0.7)'};

    &:hover {
      filter: ${!disabled && 'brightness(1.1)'};
    }
  `}
`;
