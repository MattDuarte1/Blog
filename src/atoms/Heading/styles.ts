import styled, { css, DefaultTheme } from 'styled-components';

interface ContainerProps {
  size?:
    | 'xxsmall'
    | 'xsmall'
    | 'small'
    | 'medium'
    | 'large'
    | 'xlarge'
    | 'xxlarge'
    | 'huge'
    | 'xhuge';
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue';
  uppercase: boolean;
}
const MediaFont = {
  large: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.medium};
    }
  `,
  xxlarge: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      font-size: ${theme.font.sizes.xlarge};
    }
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.large};
    }
  `,
};

const TitleSize = {
  xxsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxsmall};
  `,
  xsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${MediaFont.large(theme)};
  `,
  xlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xxlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxlarge};
    ${MediaFont.xxlarge(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.huge};
  `,
  xhuge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
};

const TitleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const TitleColor = (
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue',
  theme: DefaultTheme,
) => css`
  color: ${color ? theme.colors[color] : theme.colors.black};
`;

export const Container = styled.h1<ContainerProps>`
  ${({ theme, size, color, uppercase }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 700;
    line-height: 1.2;
    text-transform: capitalize;

    ${TitleSize[size](theme)}
    ${TitleCase(uppercase)}
    ${TitleColor(color, theme)}
  `}
`;
