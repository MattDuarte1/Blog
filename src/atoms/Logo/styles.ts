import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 600;
    font-size: ${theme.font.sizes.large};
    line-height: 45px;
    position: relative;
    &::first-letter {
      color: ${theme.colors.blue};
    }

    &::after {
      content: '.blog';
      font-family: ${theme.font.family.default};
      font-style: normal;
      font-weight: 600;
      font-size: ${theme.font.sizes.small};
      color: ${theme.colors.blue};
    }
  `}
`;
