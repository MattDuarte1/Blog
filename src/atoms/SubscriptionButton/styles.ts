import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.blue};
    border: none;
    border-radius: 25px;
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.9rem;
    padding: 1rem;

    cursor: pointer;
  `}
`;
