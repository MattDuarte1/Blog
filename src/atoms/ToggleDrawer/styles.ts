import styled, { css } from 'styled-components';

type IToggle = {
  active?: boolean;
};

export const Toggle = styled.button<IToggle>`
  ${({ theme, active }) => css`
    background-color: #fff;
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.08);
    border-radius: 10px;
    border: none;
    overflow: hidden;
    display: none;
    cursor: pointer;
    justify-content: center;
    align-items: center;
    position: relative;
    height: 70px;
    width: 70px;

    p {
      position: absolute;
      width: 40px;
      height: 4px;
      background-color: ${theme.colors.blue};
      border-radius: 4px;
      transition: 0.5s;
    }

    p:nth-child(1) {
      transform: ${active
        ? 'translateY(0px) rotate(45deg)'
        : 'translateY(-15px)'};
      width: ${active ? '40px' : '25px'};
      left: 15px;
    }

    p:nth-child(2) {
      transform: ${active
        ? 'translateY(0px) rotate(315deg)'
        : 'translateY(15px)'};
      width: ${active ? '40px' : '15px'};
      left: 15px;
    }

    p:nth-child(3) {
      transform: ${active && 'translateX(60px)'};
    }

    @media ${theme.media.lteLarge} {
      display: flex;
    }
  `}
`;
