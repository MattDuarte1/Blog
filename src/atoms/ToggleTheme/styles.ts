import styled, { css } from 'styled-components';

export const Container = styled.div`
  position: fixed;
  bottom: 5rem;
  right: 0rem;
  z-index: 2;
`;

export const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 8.9rem;
  height: 4.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0, 0, 0, 0);
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + ${Slider} {
      background: ${theme.colors.white};
    }
    &:focus + ${Slider} {
      box-shadow: 0 0 1px ${theme.colors.blue};
    }
    &:checked + ${Slider}:before {
      transform: translateX(4rem);
      background: ${theme.colors.blue};
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.lightGrey2};
    transition: all 300ms ease-in-out;
    border-radius: 2rem;
    box-shadow: 0 0 50px ${theme.colors.lightGrey2};
    &:before {
      content: '';
      position: absolute;
      height: 4.1rem;
      width: 4.1rem;
      left: 0.4rem;
      bottom: 0.2rem;
      background: ${theme.colors.blue};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
