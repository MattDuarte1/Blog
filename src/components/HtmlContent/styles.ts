import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    padding: 10rem;
    max-width: 100%;
    line-height: 1.5;

    iframe {
      width: 500px;
    }

    button {
      background-color: ${theme.colors.lightGrey2};
      padding: ${theme.spacings.xsmall};
      cursor: pointer;
      border: none;
    }

    p {
      margin: ${theme.spacings.medium} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.grey};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }
    a:hover {
      filter: brightness(50%);
    }
    code {
      font-family: monospace;
      color: ${theme.colors.black};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.darkGrey};
      padding: 0.2rem;
      margin: 0.2rem;
    }
    pre {
      background: ${theme.colors.darkGrey};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.black};
      margin: ${theme.spacings.medium} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
    pre code {
      color: inherit;
      background: inherit;
    }
    img {
      max-width: 100%;
      height: auto;
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.lightGrey1};
    }
    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }
    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 90%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
      background-color: ${theme.colors.darkGrey};
    }
    table td,
    table th {
      padding: ${theme.spacings.xsmall};
      border: 0.1rem solid ${theme.colors.lightGrey1};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.lightGrey1};
      color: ${theme.colors.darkGrey};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;
      padding: 3rem;

      iframe {
        width: 100%;
      }
    }
  `}
`;
