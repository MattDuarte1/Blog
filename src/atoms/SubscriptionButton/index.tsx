import * as Styled from './styles';
import { FaGithubAlt } from 'react-icons/fa';
import { MdClose } from 'react-icons/md';
import Heading from '@/atoms/Heading';
import { forwardRef } from 'react';
import type { Session } from 'next-auth';

export interface SubscriptionButtonProps {
  session: Session | null;
  onSubscribe: () => void;
  onUnSubscribe: () => void;
}

const SubscriptionButton = forwardRef<
  HTMLButtonElement,
  SubscriptionButtonProps
>(({ session, onSubscribe, onUnSubscribe, ...props }, ref) => {
  if (session) {
    return (
      <Styled.Container ref={ref} {...props}>
        <FaGithubAlt color="green" size={30} />
        <Heading size="xsmall">{session?.user?.name}</Heading>
        <MdClose data-testid="LogoutIcon" size={30} onClick={onUnSubscribe} />
      </Styled.Container>
    );
  }
  return (
    <Styled.Container ref={ref} onClick={onSubscribe} {...props}>
      <FaGithubAlt size={30} />
      <Heading size="xsmall">Conectar com Github</Heading>
    </Styled.Container>
  );
});

SubscriptionButton.displayName = 'Subscription Button Component';

export default SubscriptionButton;
