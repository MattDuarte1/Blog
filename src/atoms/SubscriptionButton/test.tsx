import { fireEvent, screen } from '@testing-library/react';
import type { Session } from 'next-auth';
import SubscriptionButton, { SubscriptionButtonProps } from '.';
import { renderTheme } from '@/styles/render-theme';
import { theme } from '@/styles/theme';

const mockSession: Session = {
  user: {
    name: 'mateus',
    email: 'mateus@gmail.com',
    image: '',
  },
  expires: '',
};

describe('<SubscriptionButton>', () => {
  it('Should render a SubscriptionButton when the user is not logged', () => {
    const props: SubscriptionButtonProps = {
      session: null,
      onSubscribe: () => {},
      onUnSubscribe: () => {},
    };

    renderTheme(<SubscriptionButton {...props} />);
    const subscriptionButton = screen.getByRole('button', {
      name: /conectar com github/i,
    });

    expect(subscriptionButton).toBeInTheDocument();
    expect(
      getComputedStyle(
        subscriptionButton.querySelector('h1') as HTMLHeadingElement,
      ).fontSize,
    ).toBe(theme.font.sizes.xsmall);
  });

  it('Should render a SubscriptionButton with the user name when user is logged', () => {
    const props: SubscriptionButtonProps = {
      session: mockSession,
      onSubscribe: () => {},
      onUnSubscribe: () => {},
    };

    renderTheme(<SubscriptionButton {...props} />);
    const subscriptionButton = screen.getByRole('button', { name: /mateus/i });

    expect(subscriptionButton).toBeInTheDocument();

    expect(
      getComputedStyle(
        subscriptionButton.querySelector('h1') as HTMLHeadingElement,
      ).fontSize,
    ).toBe(theme.font.sizes.xsmall);
  });

  it('Should render the Signout X Icon when the user is logged', () => {
    const props: SubscriptionButtonProps = {
      session: mockSession,
      onSubscribe: () => {},
      onUnSubscribe: () => {},
    };
    renderTheme(<SubscriptionButton {...props} />);
    const MdCloseIcon = screen.getByTestId('LogoutIcon');

    expect(MdCloseIcon).toBeInTheDocument();
    expect(MdCloseIcon).toHaveAttribute('width', '30');
  });

  it('Should call the onSubscribe function when the SubscriptionButton is clicked, but the user must be logged out.', () => {
    const handleSubscribeClicked = jest.fn();

    const props: SubscriptionButtonProps = {
      session: null,
      onSubscribe: handleSubscribeClicked,
      onUnSubscribe: () => {},
    };
    renderTheme(<SubscriptionButton {...props} />);
    const subscriptionButton = screen.getByRole('button', {
      name: /conectar com github/i,
    });

    fireEvent.click(subscriptionButton);

    expect(handleSubscribeClicked).toHaveBeenCalled();
  });

  it('Should call the onUnSubscribe function when the X Icon is clicked, but the user must be logged in.', () => {
    const handleUnSubscribeClicked = jest.fn();

    const props: SubscriptionButtonProps = {
      session: mockSession,
      onSubscribe: () => {},
      onUnSubscribe: handleUnSubscribeClicked,
    };
    renderTheme(<SubscriptionButton {...props} />);
    const MdCloseIcon = screen.getByTestId('LogoutIcon');

    fireEvent.click(MdCloseIcon);

    expect(handleUnSubscribeClicked).toHaveBeenCalled();
  });

  it('Should match snapshot SubscriptionButton when the user is not logged', () => {
    const props: SubscriptionButtonProps = {
      session: null,
      onSubscribe: () => {},
      onUnSubscribe: () => {},
    };
    renderTheme(<SubscriptionButton {...props} />);

    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('Should match snapshot SubscriptionButton when the user is logged', () => {
    const props: SubscriptionButtonProps = {
      session: mockSession,
      onSubscribe: () => {},
      onUnSubscribe: () => {},
    };
    renderTheme(<SubscriptionButton {...props} />);

    expect(screen.getByRole('button', { name: /mateus/i })).toMatchSnapshot();
  });
});
