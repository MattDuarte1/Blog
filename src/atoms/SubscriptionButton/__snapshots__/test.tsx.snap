// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionButton> Should match snapshot SubscriptionButton when the user is logged 1`] = `
.c0 {
  background: #6246EA;
  border: none;
  border-radius: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.9rem;
  padding: 1rem;
  cursor: pointer;
}

.c1 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 1.617rem;
  text-transform: none;
  color: #2B2C34;
}

<button
  class="c0"
>
  <svg
    color="green"
    fill="currentColor"
    height="30"
    stroke="currentColor"
    stroke-width="0"
    style="color: green;"
    viewBox="0 0 480 512"
    width="30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
    />
  </svg>
  <h1
    class="c1"
  >
    mateus
  </h1>
  <svg
    data-testid="LogoutIcon"
    fill="currentColor"
    height="30"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</button>
`;

exports[`<SubscriptionButton> Should match snapshot SubscriptionButton when the user is not logged 1`] = `
.c0 {
  background: #6246EA;
  border: none;
  border-radius: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.9rem;
  padding: 1rem;
  cursor: pointer;
}

.c1 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 1.617rem;
  text-transform: none;
  color: #2B2C34;
}

<button
  class="c0"
>
  <svg
    fill="currentColor"
    height="30"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 480 512"
    width="30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
    />
  </svg>
  <h1
    class="c1"
  >
    Conectar com Github
  </h1>
</button>
`;
